pkg/virtctl/set/set.go

```diff
@@ -0,0 +1,170 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 The KubeVirt Contributors
+ *
+ */
+
+package set
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"kubevirt.io/kubevirt/pkg/apimachinery/patch"
+	"kubevirt.io/kubevirt/pkg/virtctl/templates"
+)
+
+func NewSetCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "set (VM) [--cpu=CPU_COUNT] [--memory=SIZE]",
+		Short:   "Set the number of CPUs and/or memory for a running virtual machine instance.",
+		Example: setUsage(),
+		Args:    templates.ExactArgs("set", 1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return run(cmd, args, clientConfig)
+		},
+	}
+
+	cmd.Flags().Uint32("cpu", 0, "Number of CPUs (sockets) to set (must be greater than 0)")
```

Comment:
Other commands in virtctl use `Uint32Var`, `StringVar`, ... and a struct which holds the variables of a cmd.
Can you do it here too for the sake of consistency?

Reply:
i started with a struct but later tried to avoid it, because i wanted to keep things as simple as possible, anyway for the sake of consistency i will put it back